osu.Game/Utils/StatelessRNG.cs

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+namespace osu.Game.Utils
+{
+    /// Provides a fast stateless function that can be used in randomly-looking visual elements.
+    public static class StatelessRNG
+    {
+        private static ulong mix(ulong x)
+        {
+            unchecked
+            {
+                x ^= x >> 33;
+                x *= 0xff51afd7ed558ccd;
+                x ^= x >> 33;
+                x *= 0xc4ceb9fe1a85ec53;
+                x ^= x >> 33;
+                return x;
+            }
+        }
+
+        /// Compute an integer from given seed and series number.
+        /// <param name="seed">
+        /// The seed value of this random number generator.
+        /// </param>
+        /// <param name="series">
+        /// The series number.
+        /// Different values are computed for the same seed in different series.
+        /// </param>
+        public static ulong Get(int seed, int series = 0) =>
```

Comment:
I'd be strongly for returning `long`. Unsigned types are super awkward to use and always require a cast when usage is needed in signed cases (which is by far the more common case)

Reply:
If we return a negative value we cannot use it like `var face =  StatelessRNG.Get(seed) % 6`.

Remark: While taking a remainder of a RNG to interval is often discouraged due to

1. often RNG has more quality in higher-bits than lower-bits.
2. It is slightly non-uniform (getting 0 has ~ `1/(2^64/6)` higher chance than 5).

But the method is fine in this case because 1. it is opposite: the particular mix function mixes slightly mixes lower-order bits more. 2. the uniformity is maximal due to 64-bit seed, and we cannot do better.

Reply:
Maybe I should rename `ulong` version to `GetRaw` or something and make a new `GetInt(range, seed, series)`. Requiring a lengthy explanation to use is a symptom of a bad design.